### PR TITLE
Removed defaults so InstanceGroup can be defined

### PR DESCRIPTION
**Pull request by juilianoj**
AWS runJobFlow lets you use one of three options when defining your instances:
1) InstanceCount + MasterInstanceType + SlaveInstanceType
2) InstanceFleets
3) InstanceGroups

Only InstanceGroups/InstanceFleets allow you to specify volume information for your instances.  In order to define an InstanceGroup or InstanceFleets the defaults Tommy provides for InstanceCount/MasterInstanceType/SlaveInstanceType need to be removed to avoid the error of trying to use more than 1 of the methods above to define instances.

### CODE REVIEW

**Comment by mitchelldavis** [comment]
@juilianoj, The default behavior of Tommy is to allow a small cluster to spin up if there are no tommy files defined.

Instead of modifying that template file (createParameters.coffee) which would throw errors potentially.  Can we modify the default behavior to only use the default template if there are no tommy files defined?

We can accomplish this by putting an if statement around this line (https://github.com/VeriShip/tommy/blob/master/src/extender.coffee#L17).  If there are already objects in that array then there are already tommy files defined and we don't need to add the default parameters.

So, what I'm suggesting is:

* Roll back the changes you made
* Updated the extender.coffee file with the new code
* You'll need to update the following test (https://github.com/VeriShip/tommy/blob/master/tests/extender_test.coffee#L35) because it checks to see if the default parameters are merged.
* You'll probably want to write another test to check the negative of the one above too.

File: src/extender.coffee [hunk 1]
```diff
@@ -14,7 +14,8 @@ class Extender
 			if not check.object(item)
 				throw Error('You must supply an array of objects as sources.')
 
-		sources.splice(0, 0, @params)
+		if not sources.length
```
Comment (mitchelldavis) [inline src/extender.coffee:17]: @juilianoj, I'm not sure we should put `not` here.  This wouldn't pass if the length was zero would it?
